### ISSUE TITLE
memory leak

### ISSUE THREAD

**Issue by xiangnan** [zybieku/SoftKeyboardUtil#9]
第二种方式存在内存泄漏，这里

public class ActivityKeyBoardProxyBuild {
    private Activity mActivity = null;
    private ActivityKeyBoardProxyBuild() {
    }

    private static class Holder {
        private static final ActivityKeyBoardProxyBuild INSTANCE = new ActivityKeyBoardProxyBuild();
    }

    public static ActivityKeyBoardProxyBuild getInstance() {
        return Holder.INSTANCE;
    }

    public ActivityKeyBoardProxyBuild withActivity(@NonNull Activity activity) {
        this.mActivity = activity;
        return this;
    }
}

**Comment by zybieku** [comment]
@xiangnan 是的 ActivityKeyBoardProxyBuild持有了activity，改成弱引用吧，这个是伙计pr给我的，我还没看，如果大佬你觉得哪些要修改的，直接发个pr给我

**Comment by zybieku** [comment]
@Jiiiiiin

**Comment by xiangnan** [comment]
我的建议是这个类不用单例，改成普通的类，没必要单例。

**Comment by zybieku** [comment]
@xiangnan 你觉得合适，改完了发个 pr 给我，我这边很久没写android了，studio都没有

**Comment by xiangnan** [comment]
pr不会提醒么，我昨天提交了，你合一下

**Comment by zybieku** [comment]
@xiangnan 嗯 你自己测试一下 我就不测试了哈

**Comment by xiangnan** [comment]
好奇问一个不写android，换其他什么了？

**Comment by zybieku** [comment]
@xiangnan 哈哈，转Java了，后台还是稳定写，学几套框架可以好多年不用换，android跟不上了
